packages/swingset-theme-hashicorp/src/components/live-component/index.tsx

```diff
@@ -0,0 +1,43 @@
+'use client'
+import {
+  SandpackProvider,
+  SandpackLayout,
+  SandpackCodeEditor,
+  SandpackPreview,
+} from '@codesandbox/sandpack-react'
+import { Children } from 'react'
+import { getFileMap } from './get-file-map'
+import { sandpackTheme } from './code-theme'
+
+type LiveComponentProps = {
+  children: string[]
+  deps?: Record<string, string>
+}
+
+export function LiveComponent({ children, deps }: LiveComponentProps) {
+  const codeSnippets = Children.toArray(children) as React.ReactElement[]
+  const fileMap = getFileMap(codeSnippets)
+
+  const dependencies = deps ?? undefined
+
+  return (
+    <SandpackProvider
+      template="react-ts"
+      files={fileMap}
+      theme={sandpackTheme}
+      customSetup={{ dependencies }}
+    >
+      <div className="flex ss-flex-col ss-w-full ss-border-4 ss-border-gray-300 ss-rounded-lg">
+        <SandpackPreview className="ss-h-[45vh] ss-max-h-[800px] ss-min-h-[280px]" />
+        <div className="ss-bg-black ss-overflow-y-scroll">
+          <SandpackCodeEditor
+            showInlineErrors
+            showTabs
+            showRunButton
+            wrapContent
+          />
+        </div>
+      </div>
+    </SandpackProvider>
+  )
+}
```

Comment:
Using their `react-ts` because it's most flexible for our purposes, they have a template that supports app router wit `ts` however it's not stable.

Styling is temporary for POF, will talk to Ben for next steps on seeing what styles repeat themself and abstracting these into classes

Reply:
Yeah I think `react-ts` is probably the right template since we _probably_ won't have routing within a component block. That being said, I wonder if we'll have Server Components at some point? Either way, `react-ts` is a good starting point.